lib/wasi/src/ptr.rs

```diff
@@ -82,7 +82,16 @@ impl<T: Copy + ValueType> WasmPtr<T, Array> {
     }
 
     #[inline(always)]
-    pub unsafe fn get_utf8_str(self, memory: &Memory, str_len: u32) -> Option<&str> {
-        self.0.get_utf8_str(memory, str_len)
+    pub unsafe fn get_utf8_str<'a>(
+        self,
+        memory: &'a Memory,
+        str_len: u32,
+    ) -> Option<std::borrow::Cow<'a, str>> {
+        self.0.get_utf8_str(memory, str_len).map(Into::into)
+    }
+
+    #[inline(always)]
+    pub unsafe fn get_utf8_string(self, memory: &Memory, str_len: u32) -> Option<String> {
+        self.0.get_utf8_string(memory, str_len)
```

Comment:
Not tested, but I reckon it can be written as follows:

```suggestion
        self.get_utf8_str(memory, str_len)?.to_owned()
```

What do you think?